src/common/src/format.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use thiserror::Error;
+
+/// The type of format conversion to use to produce the format specifier's output.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum SpecifierType {
+    /// `s` formats the argument value as a simple string. A null value is treated as an empty
+    /// string.
+    SimpleString,
+    /// `I` treats the argument value as an SQL identifier, double-quoting it if necessary. It is
+    /// an error for the value to be null (equivalent to `quote_ident`).
+    SqlIdentifier,
+    /// `L` quotes the argument value as an SQL literal. A null value is displayed as the string
+    /// NULL, without quotes (equivalent to `quote_nullable`).
+    SqlLiteral,
+}
+
+impl TryFrom<char> for SpecifierType {
+    type Error = ();
```

Comment:
Why not returning `ParseFormatError` here?

Reply:
A part of variants are impossible here, so I don’t want to return that Error.